snuba/migrations/snuba_migrations/outcomes/0003_outcomes_add_category_and_size.py

```diff
@@ -0,0 +1,185 @@
+    Column("org_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("key_id", UInt(64)),
+    Column("timestamp", DateTime()),
+    Column("outcome", UInt(8)),
+    Column("reason", String()),
+    Column("category", UInt(8)),
+    Column("size", UInt(64)),
+    Column("times_seen", UInt(64)),
+]
+
+
+class Migration(migration.MultiStepMigration):
+    """
+    Adds the http columns defined, with the method and referer coming from the request interface
+    and url materialized from the tags.
+    """
+
+    blocking = False
+
+    def forwards_local(self) -> Sequence[operations.Operation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_raw_local",
+                column=Column("size", UInt(32, Modifiers(nullable=True))),
+                after=None,
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_raw_local",
+                column=Column("category", UInt(8)),
+                after=None,
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_hourly_local",
+                column=Column("size", UInt(64)),
+                after=None,
+            ),
+            # operations.AddColumn(
+            #     storage_set=StorageSetKey.OUTCOMES,
+            #     table_name="outcomes_hourly_local",
+            #     column=Column("category", UInt(8)),
+            #     after=""", MODIFY ORDER BY (org_id, project_id, key_id, outcome, reason, timestamp, category)""",
+            # ),
+            operations.RunSql(
+                storage_set=StorageSetKey.OUTCOMES,
+                statement="""
+            ALTER TABLE outcomes_hourly_local ADD COLUMN IF NOT EXISTS category UInt8,
+            MODIFY ORDER BY (org_id, project_id, key_id, outcome, reason, timestamp, category);
+            """,
+            ),
+            operations.DropTable(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_mv_hourly_local",
+            ),
+            operations.CreateMaterializedView(
+                storage_set=StorageSetKey.OUTCOMES,
+                view_name="outcomes_mv_hourly_local",
+                destination_table_name="outcomes_hourly_local",
+                columns=new_materialized_view_columns,
+                query="""
+                    SELECT
+                        org_id,
+                        project_id,
+                        ifNull(key_id, 0) AS key_id,
+                        toStartOfHour(timestamp) AS timestamp,
+                        outcome,
+                        ifNull(reason, 'none') AS reason,
+                        category,
+                        count() AS times_seen,
+                        ifNull(sum(size),0) AS size
```

Comment:
When can this be NULL ?

Reply:
@fpacifici agreed, this was a temp thing while working through other issues i had, it shouldn't be! 👍🏼